sycl/include/CL/sycl/types.hpp

```diff
@@ -2279,6 +2279,100 @@ __SYCL_DECLARE_FLOAT_VECTOR_CONVERTERS(double)
 #undef __SYCL_DECLARE_SCALAR_BOOL_CONVERTER
 #undef __SYCL_USE_EXT_VECTOR_TYPE__
 
+/// This macro must be defined to 1 when SYCL implementation allows user
+/// applications to explicitly declare certain class types as device copyable
+/// by adding specializations of is_device_copyable type trait class.
+#define SYCL_DEVICE_COPYABLE 1
+
+/// is_device_copyable is a user specializable class template to indicate
+/// that a type T is device copyable, which means that SYCL implementation
+/// may copy objects of the type T between host and device or between two
+/// devices.
+/// Specializing is_device_copyable such a way that
+/// is_device_copyable_v<T> == true on a T that does not satisfy all
+/// the requirements of a device copyable type may result into undefined
+/// behavior.
```

Comment:
```suggestion
/// the requirements of a device copyable type is undefined behavior.
```

Reply:
Fixed in https://github.com/intel/llvm/pull/3905/commits/ace3b783d59af3b699f9d8a0e0989a413e836b8c